homeassistant/components/device_tracker/volvooncall.py

```diff
@@ -8,32 +8,33 @@
 
 from homeassistant.util import slugify
 from homeassistant.helpers.dispatcher import (
-    dispatcher_connect, dispatcher_send)
+    async_dispatcher_connect, async_dispatcher_send)
 from homeassistant.components.volvooncall import DATA_KEY, SIGNAL_VEHICLE_SEEN
 
 _LOGGER = logging.getLogger(__name__)
 
 
-def setup_scanner(hass, config, see, discovery_info=None):
+async def async_setup_scanner(hass, config, async_see, discovery_info=None):
     """Set up the Volvo tracker."""
     if discovery_info is None:
         return
 
-    vin, _ = discovery_info
+    vin, component, attr = discovery_info
     voc = hass.data[DATA_KEY]
-    vehicle = voc.vehicles[vin]
+    instrument = voc.instrument(vin, component, attr)
 
-    def see_vehicle(vehicle):
+    async def see_vehicle(vehicle):
         """Handle the reporting of the vehicle position."""
-        host_name = voc.vehicle_name(vehicle)
+        if vehicle != vin:
+            return
+        host_name = instrument.vehicle_name
         dev_id = 'volvo_{}'.format(slugify(host_name))
-        see(dev_id=dev_id,
-            host_name=host_name,
-            gps=(vehicle.position['latitude'],
-                 vehicle.position['longitude']),
-            icon='mdi:car')
-
-    dispatcher_connect(hass, SIGNAL_VEHICLE_SEEN, see_vehicle)
-    dispatcher_send(hass, SIGNAL_VEHICLE_SEEN, vehicle)
+        await async_see(dev_id=dev_id,
+                        host_name=host_name,
+                        gps=instrument.state,
+                        icon='mdi:car')
+
+    async_dispatcher_connect(hass, SIGNAL_VEHICLE_SEEN, see_vehicle)
```

Comment:
Use new style string formatting to create a unique signal from the constant string and the vin.
```py
SIGNAL_VEHICLE_SEEN = 'volvooncall_vehicle_seen_{}'
...
signal = SIGNAL_VEHICLE_SEEN.format(vin)
```

Reply:
✓ Re-using the generic signal for state update introduced above instead.